meta/saiserialize.c

```diff
@@ -576,10 +576,38 @@ int sai_deserialize_macsec_auth_key(
         return read;
     }
 
-    SAI_META_LOG_WARN("failed to deserialize '%.*s' as macsec_auth_key", MAX_CHARS_PRINT, buffer);
+    SAI_META_LOG_WARN("failed to deserialize '%.*s' as auth_key", MAX_CHARS_PRINT, buffer);
     return SAI_SERIALIZE_ERROR;
 }
 
+int sai_serialize_macsec_sak(
+        _Out_ char *buffer,
+        _In_ const sai_macsec_sak_t sak)
+{
+   return sai_serialize_encrypt_key( buffer, sak );
+}
+
+int sai_deserialize_macsec_sak(
+        _In_ const char *buffer,
+        _Out_ sai_macsec_sak_t sak)
+{
+   return sai_deserialize_encrypt_key( buffer, sak );
```

Comment:
In the sai-ipsec subgroup meeting on July 14, it was decided to duplicate the data type for IPsec because of the unfortunate naming of the MACsec data type.
The existing sai_macsec_sak_t and the corresponding enum SAI_MACSEC_SA_ATTR_SAK should not be changed because those are used in existing MACsec direvers and NOS code.
Using sai_macsec_sak_t for IPsec could have worked with an enum SAI_IPSEC_ATTR_MACSEC_SAK, but those names have no meaning in the IPsec context.  Hence duplication was preferred.